packages/graph/hash_graph/lib/graph/src/knowledge/entity/query.rs

```diff
@@ -13,20 +13,239 @@ use crate::{
 
 #[derive(Debug, PartialEq, Eq)]
 pub enum EntityQueryPath<'q> {
+    /// The [`EntityUuid`] of the [`EntityId`] belonging to this [`Entity`].
+    ///
+    /// ```rust
+    /// # use serde::Deserialize;
+    /// # use serde_json::json;
+    /// # use graph::knowledge::EntityQueryPath;
+    /// let path = EntityQueryPath::deserialize(json!(["uuid"]))?;
+    /// assert_eq!(path, EntityQueryPath::Uuid);
+    /// # Ok::<(), serde_json::Error>(())
+    /// ```
+    ///
+    /// [`EntityUuid`]: crate::knowledge::EntityUuid
+    /// [`EntityId`]: crate::identifier::knowledge::EntityId
+    /// [`Entity`]: crate::knowledge::Entity
     Uuid,
+    /// The [`OwnedById`] of the [`EntityId`] belonging to this [`Entity`].
+    ///
+    /// ```rust
+    /// # use serde::Deserialize;
+    /// # use serde_json::json;
+    /// # use graph::knowledge::EntityQueryPath;
+    /// let path = EntityQueryPath::deserialize(json!(["ownedById"]))?;
+    /// assert_eq!(path, EntityQueryPath::OwnedById);
+    /// # Ok::<(), serde_json::Error>(())
+    /// ```
+    ///
+    /// [`OwnedById`]: crate::provenance::OwnedById
+    /// [`EntityId`]: crate::identifier::knowledge::EntityId
+    /// [`Entity`]: crate::knowledge::Entity
     OwnedById,
-    CreatedById,
-    UpdatedById,
+    /// The [`EntityVersion`] of the [`EntityEditionId`] belonging to this [`Entity`].
+    ///
+    /// ```rust
+    /// # use serde::Deserialize;
+    /// # use serde_json::json;
+    /// # use graph::knowledge::EntityQueryPath;
+    /// let path = EntityQueryPath::deserialize(json!(["version"]))?;
+    /// assert_eq!(path, EntityQueryPath::Version);
+    /// # Ok::<(), serde_json::Error>(())
+    /// ```
+    ///
+    /// [`EntityVersion`]: crate::identifier::knowledge::EntityVersion
+    /// [`EntityEditionId`]: crate::identifier::knowledge::EntityEditionId
+    /// [`Entity`]: crate::knowledge::Entity
     Version,
```

Comment:
Do we want the same note as we have for 
`DataTypeQueryPath::Version` about the "latest" keyword?